softlayer/resource_softlayer_dns_domain.go

```diff
@@ -59,14 +59,17 @@ func resourceSoftLayerDnsDomainCreate(d *schema.ResourceData, meta interface{})
 		Name: sl.String(d.Get("name").(string)),
 	}
 
-	// Create dns domain zone with default A record based on the target value
-	opts.ResourceRecords = []datatypes.Dns_Domain_ResourceRecord{
-		{
-			Data: sl.String(d.Get("target").(string)),
-			Host: sl.String("@"),
-			Ttl:  sl.Int(86400),
-			Type: sl.String("a"),
-		},
+	if targetString := sl.String(d.Get("target").(string)); *targetString != "" {
+		opts.ResourceRecords = []datatypes.Dns_Domain_ResourceRecord{
+			{
+				Data: sl.String(d.Get("target").(string)),
+				Host: sl.String("@"),
+				Ttl:  sl.Int(86400),
+				Type: sl.String("a"),
+			},
+		}
+	} else {
+		opts.ResourceRecords = []datatypes.Dns_Domain_ResourceRecord{}
```

Comment:
The default would be an empty array.
However, `json.Marshall` in `softlayer-go/session/rest.go` converts empty arrays to `null` 

Excerpt from https://golang.org/pkg/encoding/json/
```
Array and slice values encode as JSON arrays, except that []byte encodes as a base64-encoded string, and a nil slice encodes as the null JSON value
```

I tested behavior with the following sample script,
```
package main

import (
        "fmt"
        "encoding/json"
)

type OuterStruct struct {
        Name string
        Addr []InnerStruct
}

type InnerStruct struct {
        Address string
}

func main() {
        a := OuterStruct{Name: "Ivan"}
        jsonVar1, _ := json.Marshal(a)
        fmt.Printf("%s\n", jsonVar1)

        a.Addr = []InnerStruct{}
        jsonVar2, _ := json.Marshal(a)
        fmt.Printf("%s\n", jsonVar2)

}
```

Output:
```
root@ubuntuEP01:~/terraformExamples/SlTerraExample# go run nestedStructTrial.go
{"Name":"Ivan","Addr":null}
{"Name":"Ivan","Addr":[]}
```